### ISSUE TITLE
Customize compile task dependencies on javacc and jjtree tasks

### ISSUE THREAD

**Issue by QuentinBrun** [johnmartel/javaccPlugin#22]
I have a project that define multiple sourceSets but I would like that only the main sourceSet depend on javacc and jjtree tasks. 

For now, javaccPlugin add dependencies on javacc and jjtree tasks for all the compile tasks defined for all the sourceSet. 

I think that the capability to choose which compile tasks will depend on the javacc and jjtree tasks. I already wrote some code to meet my requirement and I would like to share it with you if you find this functionality interesting.

**Comment by johnmartel** [comment]
Yes, this is very interesting. Feel free to submit a pull request :)

Please try to apply the included checkstyle rules if you do so.

Thanks!

Envoyé de mon iPhone

**Comment by QuentinBrun** [comment]
Thank, I have pushed a merge request #23 . Tests and checks seems to be ok. I do not applied any checkstyle rules but I hope that it has been followed. If you have any questions, or want me to rework over something, feel free to ask. I would be very happy to see this feature added and released soon ;)

**Comment by johnmartel** [comment]
Thanks!! I will review this in the coming days.

Envoyé de mon iPhone

**Comment by JamesXNelson** [comment]
Hm.  This "helpful" wiring actually causes build breakage when plugins like jacoco are in use (and probably java9+ module builds).  Having every JavaCompile task source the output means multiple copies of classes generated and put onto classpaths.  This breaks spectacularly in the aforementioned situations.

A better fix is likely to select which sourceSet you want to add the source to, defaulting to `main`.

**Comment by zbynek** [comment]
Fixed in #65
